pkg/cluster/custom_resource.go

```diff
@@ -0,0 +1,14 @@
+package cluster
+
+import (
+	"context"
+
+	"k8s.io/client-go/dynamic"
+)
+
+type CustomConnection interface {
+	CreateCustomResource(ctx context.Context, c *KubernetesCluster, serviceID string) error
+	CustomResourceExists(ctx context.Context, c *KubernetesCluster, serviceID string) error
+	BindCustomConnection(ctx context.Context, serviceName string, options ServiceBindingOptions, clients *KubernetesClients) error
+	CustomConnectionExists(ctx context.Context, dynamicClient dynamic.Interface, serviceName string, ns string) error
```

Comment:
It seems CustomConnectionExists and CustomResourceExists tend to do the same thing in different way, will it make sense to have one method for it?

Reply:
yes